server/src/main/java/org/loomdev/loom/entity/player/PlayerImpl.java

```diff
@@ -157,10 +157,14 @@ public void sendTitle(@NotNull Component title, @NotNull Component subtitle, int
     }
 
     @Override
-    @Nullable
-    public InetSocketAddress getRemoteAddress() {
-        if (!isConnected()) return null;
-        return (InetSocketAddress) getMinecraftEntity().connection.connection.getRemoteAddress();
+    @NotNull
+    public Optional<InetSocketAddress> getRemoteAddress() {
+        if (isConnected()) {
+            return Optional.of(getMinecraftEntity().connection.connection.getRemoteAddress())
+                    .map(InetSocketAddress.class::cast);
+        }
+
+        return Optional.empty();
```

Comment:
I would reverse the return locations, so the longer line has the least indentation. (Improved readability)